From 58975d82e947e67e96eac098f70290e813071374 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:44 -0400
Subject: [ethernet] mlx5: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <8e71f419efcd74e0de64ad1ff2fa1d26da6a6bf9.1410425016.git.avadai@redhat.com>
Patchwork-id: 93611
O-Subject: [Patch RHEL7.1 V1 048/197] mlx5: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range() and pci_enable_msix_range()
interfaces.

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Cc: Eli Cohen <eli@mellanox.com>
Cc: linux-rdma@vger.kernel.org
Cc: netdev@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit f3c9407bc201f2e3f0e6845e9921454945568ed1)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/main.c b/drivers/net/ethernet/mellanox/mlx5/core/main.c
index a064f06..81df046 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/main.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/main.c
@@ -116,7 +116,6 @@ static int mlx5_enable_msix(struct mlx5_core_dev *dev)
  struct mlx5_eq_table *table = &dev->priv.eq_table;
  int num_eqs = 1 << dev->caps.log_max_eq;
  int nvec;
- int err;
  int i;
 
  nvec = dev->caps.num_ports * num_online_cpus() + MLX5_EQ_VEC_COMP_BASE;
@@ -131,17 +130,12 @@ static int mlx5_enable_msix(struct mlx5_core_dev *dev)
  for (i = 0; i < nvec; i++)
   table->msix_arr[i].entry = i;
 
-retry:
- table->num_comp_vectors = nvec - MLX5_EQ_VEC_COMP_BASE;
- err = pci_enable_msix(dev->pdev, table->msix_arr, nvec);
- if (err <= 0) {
-  return err;
- } else if (err > 2) {
-  nvec = err;
-  goto retry;
- }
+ nvec = pci_enable_msix_range(dev->pdev, table->msix_arr,
+         MLX5_EQ_VEC_COMP_BASE, nvec);
+ if (nvec < 0)
+  return nvec;
 
- mlx5_core_dbg(dev, "received %d MSI vectors out of %d requested\n", err, nvec);
+ table->num_comp_vectors = nvec - MLX5_EQ_VEC_COMP_BASE;
 
  return 0;
 }
-- 
1.7.1